1-mandatory/2-quotegenerator/quotes.js

```diff
@@ -1,492 +1,514 @@
-  },
-  {
-    quote: "Dreaming, after all, is a form of planning.",
-    author: "Gloria Steinem",
-  },
-  {
-    quote:
-      "It’s your place in the world; it’s your life. Go on and do all you can with it, and make it the life you want to live.",
-    author: "Mae Jemison",
-  },
-  {
-    quote:
-      "You may be disappointed if you fail, but you are doomed if you don’t try.",
-    author: "Beverly Sills",
-  },
-  {
-    quote: "Remember no one can make you feel inferior without your consent.",
-    author: "Eleanor Roosevelt",
-  },
-  {
-    quote: "Life is what we make it, always has been, always will be.",
-    author: "Grandma Moses",
-  },
-  {
-    quote:
-      "The question isn’t who is going to let me; it’s who is going to stop me.",
-    author: "Ayn Rand",
-  },
-  {
-    quote:
-      "When everything seems to be going against you, remember that the airplane takes off against the wind, not with it.",
-    author: "Henry Ford",
-  },
-  {
-    quote:
-      "It’s not the years in your life that count. It’s the life in your years.",
-    author: "Abraham Lincoln",
-  },
-  {
-    quote: "Change your thoughts and you change your world.",
-    author: "Norman Vincent Peale",
-  },
-  {
-    quote:
-      "Either write something worth reading or do something worth writing.",
-    author: "Benjamin Franklin",
-  },
-  {
-    quote: "Nothing is impossible, the word itself says, “I’m possible!”",
-    author: "–Audrey Hepburn",
-  },
-  {
-    quote: "The only way to do great work is to love what you do.",
-    author: "Steve Jobs",
-  },
-  {
-    quote: "If you can dream it, you can achieve it.",
-    author: "Zig Ziglar",
-  },
-];
+// DO NOT EDIT BELOW HERE
+
+// A function which will return one item, at
+// random, from the given array.
+//
+// Parameters
+// ----------
+// choices: an array of items to pick from.
+//
+// Returns
+// -------
+// One item of the given array.
+//
+// Examples of use
+// ---------------
+// pickFromArray([1,2,3,4])     //maybe returns 2
+// pickFromArray(coloursArray)  //maybe returns "#F38630"
+//
+// You DO NOT need to understand how this function works.
+
+
+
+/* code to generate random quotes*/
+  // get the element where u want to show the quote
+  const quoteElement = document.querySelector(".quote");
+  //  get the element to show author name
+const authorElemnt = document.querySelector(".author");
+// get the btn of New Quote
+const btn = document.getElementById("quote-btn");
+//  add event listener to the btn
+btn.addEventListener("click", genearteQuote)
+//  declare the function generateQuote
+function genearteQuote(){
+  // getting the value of generateRandomQuotes()
+  let randomQuote = generateRandomQuotes ()
+// adding text to the QuoteEklement & authorElement  where the codes will be shown 
+quoteElement.innerHTML =  quotes[randomQuote].quote
+authorElemnt.innerHTML = quotes[randomQuote].author
+}
+
+// generate random quotes
+function generateRandomQuotes() {

```

Comment:
a better name for this function might be `generateRandomIndex` - as it's not actually returning a quote, just an index